src/main/java/io/lettuce/core/models/events/CommandSucceededEvent.java

```diff
@@ -0,0 +1,31 @@
+package io.lettuce.core.models.events;
+
+import java.time.Duration;
+import java.time.Instant;
+
+/**
+ * @author sokomishalov
+ */
+public class CommandSucceededEvent<K, V, T> extends CommandBaseEvent<K, V, T> {
+
+    private final Instant startedAt;
+    private final Instant succeededAt;
+
+    public CommandSucceededEvent(CommandStartedEvent<K, V, T> startedEvent) {
+        super(startedEvent.getCommand(), startedEvent.getContext());
+        this.startedAt = startedEvent.getStartedAt();
+        this.succeededAt = Instant.now();
+    }
+
+    public Duration getExecuteDuration() {
+        return Duration.between(startedAt, succeededAt);
+    }
+
+    public Instant getStartedAt() {
+        return startedAt;
```

Comment:
Here's a good spot to create an `instant` from a timestamp.

Reply:
done